plonky2/src/plonk/circuit_data.rs

```diff
@@ -94,6 +94,10 @@ impl CircuitConfig {
     }
 
     pub fn standard_recursion_zk_config() -> Self {
+        #[cfg(not(feature = "rand"))]
```

Comment:
I don't think it's useful to have another assert here, the one you added in `check_config` seems like the best place IMO (it'll give early detection and it's in a code path that's always hit).